src/lib/autoinstall/clients/clone_system.rb

```diff
@@ -151,9 +152,11 @@ def clone_system(options)
             # always clone general
             Yast::ProductControl.clone_modules + ["general"]
           end
+
         Yast::AutoinstClone.Process(target: target)
         begin
           Yast::XML.YCPToXMLFile(:profile, Yast::Profile.current, filename)
+          ::FileUtils.chmod(0o600, filename)
```

Comment:
Unfortunately there is still a small race condition, the file will still have the default permission for short amount of time. Moreover if an exception occurs during writing this `chmod` will be skipped.

Maybe a solution would be to first create an empty file with the required permissions and then let `YCPToXMLFile()` overwrite the file.